### ISSUE TITLE
Feature request: copy locally-declared variables in generated `<script>`

### ISSUE THREAD

**Issue by innocenzi** [sxzz/unplugin-vue-define-options#2]
First of all, thanks! This is a really needed feature. especially for third-party Vue plugins.

https://github.com/sxzz/unplugin-vue-define-options/blob/main/src/core/utils.ts#L55

I understand the restriction, but I think it could potentially be worked around. In my case, I am using [Inertia](https://inertiajs.com) and I want to use a persistent layout.

If you are not familiar with Inertia, we need to declare a `layout` property on the Vue component itself:

```vue
<script lang="ts">
import layout from '@/views/layouts/default.vue'

export default {
  layout
}
</script>

<script setup lang="ts">
// The rest of our code
</script>
```

I already have a "solution" for this with a Vite plugin: https://twitter.com/enzoinnocenzi/status/1443171187190808578

However, I think `defineOptions` is also a very good potential alternative for this feature and I would like to use it:

```vue
<script setup lang="ts">
import layout from '@/views/layouts/default.vue'

defineOptions({
  layout,
})

// The rest of our code
</script>
```

This issue is that the macro doesn't support local variables, so I can't import the layout there. Do you think this could be worked around by copying local variables/imports to the generated `<script>`?

**Comment by sxzz** [comment]
It now supports referenced import variables. Please upgrade to v0.2.0 and try again! 😆

**Comment by innocenzi** [comment]
Thanks, that was super fast! Unfortunately it seems that it does not work properly. :(

I tested with the following code:

```vue
<script setup lang="ts">
import contained from '../layouts/contained.vue'

defineOptions({
	layout: contained,
})
</script>
```

Here is the transformation applied by the plugin:

![](https://i.imgur.com/ZJOWXAN.png)

The transformed output should be something like:

```vue
<script lang="ts">
import contained from '../layouts/contained.vue'

export default {
	layout: contained,
}
</script>
<script setup lang="ts">
</script>
```

**Comment by sxzz** [comment]
I can't reproduce it. Would you like share a minimum reproduction repo for me?

**Comment by innocenzi** [comment]
Sure, here it is: https://github.com/innocenzi/define-options-issue-2

```sh
git clone innocenzi/define-options-issue-2
cd define-options-issue-2
pnpm i
pnpm run dev
```

![](https://i.imgur.com/13SKncS.png)

**Comment by sxzz** [comment]
Released as v0.2.1. It should be fixed.

**Comment by innocenzi** [comment]
It looks like it works, but sometimes I get a weird error, which I suspect is HMR-related. I'll investigate and open a separate issue. In the meantime, thanks a lot for that feature!
